src/leap/mail/imap/mailbox.py

```diff
@@ -457,7 +457,33 @@ def expunge(self):
             raise imap4.ReadOnlyMailbox
         return self.collection.delete_all_flagged()
 
-    def _bound_seq(self, messages_asked):
+    def _get_message_fun(self, uid):
+        """
+        Return the proper method to get a message for this mailbox, depending
+        on the passed uid flag.
+
+        :param uid: If true, the IDs specified in the query are UIDs;
+                    otherwise they are message sequence IDs.
+        :type uid: bool
+        :rtype: callable
+        """
+        get_message_fun = [
+            self.collection.get_message_by_sequence_number,
+            self.collection.get_message_by_uid][uid]
```

Comment:
I'm not sure how much I like using a bool as an array index. But if you like your hack I'm happy to keep it here.